modules/loaders-common/src/main/scala/com/snowplowanalytics/snowplow/loaders/transform/SchemaProvider.scala

```diff
@@ -12,48 +12,28 @@ import cats.effect.Sync
 import cats.implicits._
 import com.snowplowanalytics.iglu.client.{ClientError, Resolver}
 import com.snowplowanalytics.iglu.client.resolver.registries.RegistryLookup
-import com.snowplowanalytics.iglu.core.{SchemaCriterion, SchemaKey, SchemaMap, SelfDescribingSchema}
+import com.snowplowanalytics.iglu.client.resolver.Resolver.SchemaResolutionError
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaMap, SelfDescribingSchema}
 import com.snowplowanalytics.iglu.schemaddl.jsonschema.Schema
 import com.snowplowanalytics.iglu.schemaddl.jsonschema.circe.implicits.toSchema
 import com.snowplowanalytics.snowplow.badrows.FailureDetails
 
-import scala.math.Ordered._
-
 private[transform] object SchemaProvider {
 
-  private def getSchema[F[_]: Sync: RegistryLookup](
-    resolver: Resolver[F],
-    schemaKey: SchemaKey
-  ): EitherT[F, FailureDetails.LoaderIgluError, SelfDescribingSchema[Schema]] =
-    for {
-      json <- EitherT(resolver.lookupSchema(schemaKey))
-                .leftMap(resolverBadRow(schemaKey))
-      schema <- EitherT.fromOption[F](Schema.parse(json), parseSchemaBadRow(schemaKey))
-    } yield SelfDescribingSchema(SchemaMap(schemaKey), schema)
-
   // Note schema order of the returned list is not guaranteed
   def fetchSchemasWithSameModel[F[_]: Sync: RegistryLookup](
     resolver: Resolver[F],
     schemaKey: SchemaKey
   ): EitherT[F, FailureDetails.LoaderIgluError, NonEmptyVector[SelfDescribingSchema[Schema]]] =
     for {
-      schemaKeys <- EitherT(resolver.listSchemasLike(schemaKey))
-                      .leftMap(resolverFetchBadRow(schemaKey.vendor, schemaKey.name, schemaKey.format, schemaKey.version.model))
-                      .map(_.schemas.toVector)
-      schemaKeys <- EitherT.rightT[F, FailureDetails.LoaderIgluError](schemaKeys.filter(_ < schemaKey))
-      topSchema <- getSchema(resolver, schemaKey)
-      lowerSchemas <- schemaKeys.filter(_ < schemaKey).traverse(getSchema(resolver, _))
-    } yield NonEmptyVector(topSchema, lowerSchemas)
-
-  private def resolverFetchBadRow(
-    vendor: String,
-    name: String,
-    format: String,
-    model: Int
-  )(
-    e: ClientError.ResolutionError
-  ): FailureDetails.LoaderIgluError =
-    FailureDetails.LoaderIgluError.SchemaListNotFound(SchemaCriterion(vendor = vendor, name = name, format = format, model = model), e)
```

Comment:
Hi @istreeter , with the new function we know precisely for which schema there was an issue. Rather than keeping the existing `SchemaListNotFound(SchemaCriterion)` it seems to me that it's better to just return the faulty schema, because it provides more info for the troubleshooting, and we know the schema criterion related to the error anyway. WDYT ?

Reply:
Yeah I agree